Zend/tests/arrow_functions/001.phpt

```diff
@@ -3,43 +3,87 @@ Basic arrow function functionality check
 --FILE--
 <?php
 
+// No return value
+$foo = fn() => {};
```

Comment:
I wonder, why `fn() => {}` rather than just `fn() {}`?

Reply:
also what about `fn() use (&$byRef) {}`?

Reply:
While it may be consistent with other languages, it is more important to be internally consistent within PHP. Especially when I view this in conjunction with #6221, which introduces `=> expr` for method bodies, it seems like we should be interpreting `=> expr;` as a shorthand for `{ return expr; }`. In this context `=> { stmt; }` does not make sense syntactically.

Of course, there is also an alternative view here: If we add general support for block expressions, then the syntax `=> { stmt; }` *would* make sense. But that would have larger implications, in that it should also be usable outside the context of arrow functions. You will want to read up on the discussions we had on this when considering block expressions for `match`.

Reply:
Totally agree on your point here. I was not aware of #6221, but it does in fact make sense. I would consider your second point as some sort of block expression like Rust have. Could you point me to the discussion regarding block expressions? It peaked my curiosity ;)

Reply:
It's a bit spread out, https://externals.io/message/109941#109947, #5403, #5448 seem relevant, there's probably more in the other match discussion threads.

Reply:
@nikic I will make sure read those discussions before making an RFC. It may take a while.

Reply:
I think '=>' is not necessary since PHP has the 'fn' keyword

Reply:
Addressed here: https://github.com/php/php-src/pull/6246/commits/b82fe0c8a304a068fc0af90aa4b2b2cf14c66002.